pkg/controller/migcluster/remote_watch_map.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2019 Red Hat Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package migcluster
+
+import (
+	"sync"
+
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+var rwmInstance *RemoteWatchMap
+var createRemoteWatchMapOnce sync.Once
+
+// RemoteWatchCluster tracks Remote Managers and Event Forward Channels
+type RemoteWatchCluster struct {
+	ForwardChannel chan event.GenericEvent
+	RemoteManager  manager.Manager
+	//  TODO - setup stop channel for manager so that manager will stop when we close the event channel from parent
+}
+
+// RemoteWatchMap provides a map between MigCluster nsNames and RemoteWatchClusters
+type RemoteWatchMap struct {
+	sync.RWMutex
```

Comment:
nesting the `RWMutex` will make it's methods public on the `RemoteWatchMap`.  It would be safer to add like:
```
mutex sync.RWMutex
```
the do:
```
r.mutex.RLock()
defer r.mutex.RUnlock()
```
Basically, having the object's mutex be private is safer.  I don't think we'd every want something external to access the mutex.

That said, I don't know what is customary for _Go_`.

Reply:
ACK